src/main/java/com/google/codeu/servlets/FormHandlerServlet.java

```diff
@@ -26,6 +33,29 @@
 @WebServlet("/my-form-handler")
 public class FormHandlerServlet extends HttpServlet {
 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+    ServletOutputStream out = response.getOutputStream();
+    out.println("<h1>Here is the post you uploaded</h1>");
+    out.println("<ul>");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
This block of new code (43-54) breaks a concept in OOP known as encapsulation, the gist of which is that any interaction with DatastoreService should be kept within Datastore.java. The purpose of Datastore is to provide a layer of abstraction over the actual implementation of data access. The rest of the codebase doesn't have to know about DatastoreService or Queries or Entities because Datastore hides all of those details. The rest of the code just knows that Datastore will store and retrieve Message objects that are much friendlier to work with.

The TL;DR of why you do this is it makes changing data access implementations much easier. If one day you decide that instead of DatastoreService, you want to work with MySQL or Firebase or whatever the cool new database tech of the time is, all you have to do is rewrite the method implementations in Datastore. As far as the rest of the code is concerned, nothing has changed because they still only know that there is a class called Datastore that does data stuff for them.

A similar concept of encapsulation should be applied here, where you could either:
1) Create a new class called ImageUrl or Post or something has fields for the URL, message, and location, then write methods in Datastore that convert between an object of that new class into Entities and Queries (similar to the existing methods for Message). Use those Datastore methods here instead of directly accessing DatastoreService.
2) Modify the existing Message class and the associated methods in Datastore.java. As far as I can tell, these image posts fulfill very similar functionality to Messages in the original project framework (users post messages, users can see posted messages), and we don't really even use Messages anymore.